### PR TITLE
chore(tags): migrate Tag examples to storybook

### PR DESCRIPTION
**Pull request by austingreendev**
## Description

This PR migrates the `react-tags` documentation to Storybook.

## Checklist

- [ ] ~:ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)~
- [x] :globe_with_meridians: demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :metal: renders as expected with Bedrock CSS (`?bedrock`)
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [ ] ~:guardsman: includes new unit tests~
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: packages/tags/stories/examples/Advanced.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+interface IAdvancedStoryProps {
+  size: 'small' | 'medium' | 'large';
+  isPill: boolean;
+  includeAvatar: boolean;
+  includeClose: boolean;
+  width: number;
+}
+
+export const Advanced: Story<IAdvancedStoryProps> = ({
```
Comment (jzempel) [inline packages/tags/stories/examples/Advanced.tsx:33]: This story no longer demonstrates the functionality from the Styleguidist example. https://zendeskgarden.github.io/react-components/tags/#advanced
This is an important visual check to retain here.

File: packages/tags/stories/examples/CustomHue.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import { Story } from '@storybook/react';
+import { action } from '@storybook/addon-actions';
+import { Tag } from '@zendeskgarden/react-tags';
+import { Col, Grid, Row } from '@zendeskgarden/react-grid';
+
+interface ICustomHueStoryProps {
+  hue: string;
+}
+
+export const CustomHue: Story<ICustomHueStoryProps> = ({ hue }) => (
```
Comment (jzempel) [inline packages/tags/stories/examples/CustomHue.tsx:18]: This story no longer demonstrates the previous Styleguidist example: https://zendeskgarden.github.io/react-components/tags/#!/Advanced/3

File: packages/tags/stories/examples/Default.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+          hue={hue === 'default' ? undefined : hue}
+          isPill={shape === 'pill'}
+          isRound={shape === 'round'}
+          isRegular={isRegularWeight}
+          size={size}
+          tabIndex={0}
+        >
+          {includeAvatar && (
+            <Tag.Avatar>
+              <img alt="" src={`images/avatar-${Math.floor(Math.random() * 70 + 1)}.png`} />
```
Comment (jzempel) [inline packages/tags/stories/examples/Default.tsx:46]: Let's simplify the code (and deployment size) by selecting a single, fixed avatar.

**Review by jzempel — CHANGES_REQUESTED** [review]

File: packages/tags/stories/examples/FontWeight.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      </Col>
+    </Row>
+  </Grid>
+);
+
+FontWeight.parameters = {
+  docs: {
+    storyDescription: `
+The following examples demonstrates using the \`isRegular\` prop combined with
+mixed weight content. Check out the code for details.
```
Comment (jzempel) [inline packages/tags/stories/examples/FontWeight.tsx:42]: The text indicates that I can check out the code, but the "Show code" button is disabled.

**Review by jzempel — CHANGES_REQUESTED** [review]

File: packages/tags/stories/examples/CustomHue.tsx (hunk 2 above)
Comment (austingreendev) [reply packages/tags/stories/examples/CustomHue.tsx:18]: This example includes the single hue color picker.

image

What other controls are missing?

File: packages/tags/stories/examples/Advanced.tsx (hunk 1 above)
Comment (austingreendev) [reply packages/tags/stories/examples/Advanced.tsx:33]: I've included the `width`, `size`, `pill`, `Tag.Avatar`, and `Tag.Close` options. What other controls need to be provided?

image

File: packages/tags/stories/examples/Default.tsx (hunk 3 above)
Comment (austingreendev) [reply packages/tags/stories/examples/Default.tsx:46]: Updated to a single image.

File: packages/tags/stories/examples/FontWeight.tsx (hunk 4 above)
Comment (austingreendev) [reply packages/tags/stories/examples/FontWeight.tsx:42]: Was able to track it down to this bug https://github.com/storybookjs/storybook/issues/13362

I've updated this story as well as 2 "No code available" failures in the `Avatar` examples.

File: packages/tags/stories/examples/Advanced.tsx (hunk 1 above)
Comment (jzempel) [reply packages/tags/stories/examples/Advanced.tsx:33]: Oh geez, I can't storybook 🤦 I was on the "Docs" tab.

File: packages/tags/stories/examples/CustomHue.tsx (hunk 2 above)
Comment (jzempel) [reply packages/tags/stories/examples/CustomHue.tsx:18]: My bad.

**Review by jzempel — APPROVED** [review]
[nit] the "Advanced" story would probably read better if it were named "Truncation".

**Comment by austingreendev** [comment]
Updated to "Truncation". Much better now.

**Pull request merged**